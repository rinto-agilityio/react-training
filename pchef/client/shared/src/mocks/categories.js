export const categories = [
  {
    id: 'chicken',
    name: 'Chicken',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Heathy',
    name: 'Heathy',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Breakast',
    name: 'Easy Breakast',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Dessert',
    name: 'Dessert',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Fish',
    name: 'Fish',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Pizza',
    name: 'Pizza',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Cake',
    name: 'Cake',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Deserts',
    name: 'Deserts',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Salad',
    name: 'Salad',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Kids',
    name: 'Kids',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Meat',
    name: 'Meat',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Lunch',
    name: null,
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
]
