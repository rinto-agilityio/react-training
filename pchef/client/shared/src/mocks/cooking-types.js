export const cookingTypes = [
  {
    id: 'baking',
    name: 'Baking',
  },
  {
    id: 'roasting',
    name: 'Roasting',
  },
  {
    id: 'frying',
    name: 'Frying',
  },
]
