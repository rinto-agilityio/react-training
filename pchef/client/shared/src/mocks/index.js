// mock categories
export const category = {
  name: 'Heathy',
  imgUrl:
    'https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
}
