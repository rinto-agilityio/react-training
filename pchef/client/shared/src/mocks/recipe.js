export const recipe = {
  description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
  title: 'Bun Bo Hue Recipe',
  imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
  votes: [1],
  steps: [
    {
      step: 1,
      description: 'Step 1 description',
      title: 'Step 1 title',
      subTitle: 'Step 1 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 2,
      description: 'Step 2 description',
      title: 'Step 2 title',
      subTitle: 'Step 2 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 3,
      description: 'Step 3 description',
      title: 'Step 3 title',
      subTitle: 'Step 3 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 4,
      description: 'Step 4 description',
      title: 'Step 4 title',
      subTitle: 'Step 4 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 5,
      description: 'Step 5 description',
      title: 'Step 5 title',
      subTitle: 'Step 5 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 6,
      description: 'Step 6 description',
      title: 'Step 6 title',
      subTitle: 'Step 6 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
  ],
}
