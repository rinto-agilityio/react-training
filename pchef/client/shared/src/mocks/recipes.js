export const recipes = [
  {
    id: 0,
    title: 'Bun Bo Hue Recipe',
    subTitle: 'Traditional food',
    description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
    imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    votes: [1],
    views: 1,
    steps: [
      {
        step: 1,
        description: 'Step 1 description',
        title: 'Step 1 title',
        subTitle: 'Step 1 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 2,
        description: 'Step 2 description',
        title: 'Step 2 title',
        subTitle: 'Step 2 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 3,
        description: 'Step 3 description',
        title: 'Step 3 title',
        subTitle: 'Step 3 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 4,
        description: 'Step 4 description',
        title: 'Step 4 title',
        subTitle: 'Step 4 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 5,
        description: 'Step 5 description',
        title: 'Step 5 title',
        subTitle: 'Step 5 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 6,
        description: 'Step 6 description',
        title: 'Step 6 title',
        subTitle: 'Step 6 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
    ],
  },
  {
    id: 1,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 2,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 3,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
]
