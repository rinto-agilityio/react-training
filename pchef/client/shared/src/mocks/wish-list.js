export const wishList = [{
  id: '1',
  date: Date.now(),
  categoryId: '1',
  cookingTypeId: '1',
},
{
  id: '2',
  date: Date.now(),
  categoryId: '2',
  cookingTypeId: '2',
}]
