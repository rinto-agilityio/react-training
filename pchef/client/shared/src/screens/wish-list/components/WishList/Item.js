// @flow
// add flow above to fix for using flow with React.memo

// Libs
import React, { memo } from 'react'
import { Text, View } from 'react-native'
import whyDidYouRender from '@welldone-software/why-did-you-render'

// Styles
import styles from './styles'

// Helpers
import { getDateForCalendar } from '../../../../helpers/date-time'

import type { WishList } from '../../../../flow-types/wish-list'
import type { Category } from '../../../../flow-types/category'
import type { CookingType } from '../../../../flow-types/cooking-type'

if (process.env.NODE_ENV !== 'production') {
  whyDidYouRender(React, {
    onlyLogs: true,
    titleColor: 'green',
    diffNameColor: 'aqua',
  })
}

type Props = {
  size: string,
  item: WishList,
  categories: Array<Category>,
  cookingTypes: Array<CookingType>,
}

const Item = ({
  size = 'medium',
  item = {},
  categories = [],
  cookingTypes = [],
}: Props) => {
  const {
    id,
    date,
    categoryId,
    cookingTypeId,
  } = item

  const category = categories.find(item => item.id === categoryId) || {}
  const cookingType = cookingTypes.find(item => item.id === cookingTypeId) || {}

  return (
    (category.name && cookingType.name) ? (
      <View
        style={[styles.wrapperItem, styles[`${size}WrapperItem`]]}
        key={id}
      >
        <Text style={[styles.date, styles[`${size}Date`]]}>{getDateForCalendar(Number(date))}</Text>
        <Text style={[styles.content, styles[`${size}Content`]]}>
          Get the recipe for
          <Text
            style={[
              styles.content,
              styles[`${size}Content`],
              styles.specialContent]
            }
          >
            {` ${category.name} `}
          </Text>
          with
          <Text
            style={[
              styles.content,
              styles[`${size}Content`],
              styles.specialContent]
            }
          >
            {` ${cookingType.name} `}
          </Text>
        </Text>
      </View>
    ) : null
  )
}

Item.whyDidYouRender = true

export default memo<Props>(Item)
