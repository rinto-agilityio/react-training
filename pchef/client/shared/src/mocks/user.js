import { categories } from './categories'
import { recipes } from './recipes'

const favoriteRecipeId: Array<string> = recipes.map(item => item.id)
export const user = {
  id: '1',
  name: 'Alex',
  email: 'alex@gmail.com',
  avatar:
    'https://cdn4.iconfinder.com/data/icons/avatars-21/512/avatar-circle-human-female-black-7-512.png',
  favoriteRecipe: favoriteRecipeId,
  followCategory: categories,
}
