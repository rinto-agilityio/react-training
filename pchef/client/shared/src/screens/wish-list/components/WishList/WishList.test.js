// Components
import WistList from '.'
import Item from './Item'

// Mocks
import { wishList, categories, cookingTypes } from '../../../../mocks'

describe('Wish list', () => {
  let wishListProps = {
    wishList,
  }

  let wishProps = {
    item: wishList[0],
  }

  it('Renders correctly wish list commponent', () => {
    const wishList = renderer.create(<WistList {...wishListProps} />).toJSON()
    expect(wishList).toMatchSnapshot()
  })

  it('Renders correctly wish commponent', () => {
    const wish = renderer.create(<Item {...wishProps} />).toJSON()
    expect(wish).toMatchSnapshot()
  })

  it('Render wish list component with size medium', () => {
    wishListProps = {
      ...wishProps,
      size: 'medium',
    }
    const wishList = renderer.create(<WistList {...wishListProps} />).toJSON()
    expect(wishList).toMatchSnapshot()
  })

  it('Render wish list component with loading true', () => {
    wishListProps = {
      ...wishProps,
      loading: true,
    }
    const wishList = renderer.create(<WistList {...wishListProps} />).toJSON()
    expect(wishList).toMatchSnapshot()
  })

  it('Render wish list component with error', () => {
    wishListProps = {
      ...wishProps,
      error: {
        graphQLErrors: [{
          message: 'Error!',
        }],
      },
    }
    const wishList = renderer.create(<WistList {...wishListProps} />).toJSON()
    expect(wishList).toMatchSnapshot()
  })

  it('Render wish component with size medium', () => {
    wishProps = {
      ...wishProps,
      size: 'medium',
    }
    const wish = renderer.create(<Item {...wishProps} />).toJSON()
    expect(wish).toMatchSnapshot()
  })

  it('Render wish component with category, cookingType', () => {
    wishProps = {
      ...wishProps,
      categories,
      cookingTypes,
    }
    const wish = renderer.create(<Item {...wishProps} />).toJSON()
    expect(wish).toMatchSnapshot()
  })

  it('Render wish list component with defaultProps', () => {
    wishListProps = {}
    const wishList = renderer.create(<WistList {...wishListProps} />).toJSON()
    expect(wishList).toMatchSnapshot()
  })

  it('Render wish component with defaultProps', () => {
    wishProps = {}
    const wish = renderer.create(<Item {...wishProps} />).toJSON()
    expect(wish).toMatchSnapshot()
  })
})
