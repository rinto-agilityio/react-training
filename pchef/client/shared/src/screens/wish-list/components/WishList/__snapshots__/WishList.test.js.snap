// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wish list Render wish component with category, cookingType 1`] = `
<div
  className="css-view-1dbjc4n r-borderBottomColor-1nd8m0h r-borderBottomWidth-qklmqi r-display-6koalj r-padding-1pcd2l5"
>
  <div
    className="css-text-76zvg2 r-color-cqee49 r-fontSize-adyw6z r-fontWeight-vw2c0b r-marginBottom-117bsoe"
    dir="auto"
  >
    2019-05-28
  </div>
  <div
    className="css-text-76zvg2 r-fontSize-1i10wst"
    dir="auto"
  >
    Get the recipe for
    <span
      className="css-text-76zvg2 css-textHasAncestor-16my406 r-fontSize-1i10wst r-fontWeight-vw2c0b"
      dir="auto"
    >
       Heathy 
    </span>
    with
    <span
      className="css-text-76zvg2 css-textHasAncestor-16my406 r-fontSize-1i10wst r-fontWeight-vw2c0b"
      dir="auto"
    >
       Roasting 
    </span>
  </div>
</div>
`;

exports[`Wish list Render wish component with defaultProps 1`] = `null`;

exports[`Wish list Render wish component with size medium 1`] = `null`;

exports[`Wish list Render wish list component with defaultProps 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;

exports[`Wish list Render wish list component with error 1`] = `
<div
  className="css-text-76zvg2 r-color-1vs7qoz r-fontSize-1i10wst r-fontStyle-36ujnk"
  dir="auto"
>
  Error!
</div>
`;

exports[`Wish list Render wish list component with loading true 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-drfeu3 r-bottom-1p0dtai r-justifyContent-1777fci r-left-1d2f490 r-position-1xcajam r-right-zchlnj r-top-ipm5af r-zIndex-1sg8ghl"
>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
    role="progressbar"
  >
    <div
      className="css-view-1dbjc4n r-animationDuration-17bb2tj r-animationIterationCount-1muvv40 r-animationKeyframes-127358a r-animationTimingFunction-1ldzwu0 r-height-z80fyv r-width-19wmn03"
    >
      <svg
        height="100%"
        viewBox="0 0 32 32"
        width="100%"
      >
        <circle
          cx="16"
          cy="16"
          fill="none"
          r="14"
          strokeWidth="4"
          style={
            Object {
              "opacity": 0.2,
              "stroke": "#c5c5c5",
            }
          }
        />
        <circle
          cx="16"
          cy="16"
          fill="none"
          r="14"
          strokeWidth="4"
          style={
            Object {
              "stroke": "#c5c5c5",
              "strokeDasharray": 80,
              "strokeDashoffset": 60,
            }
          }
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Wish list Render wish list component with size medium 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;

exports[`Wish list Renders correctly wish commponent 1`] = `null`;

exports[`Wish list Renders correctly wish list commponent 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;
